draft-ietf-quic-qpack.md

```diff
@@ -456,31 +458,36 @@ referenced by a given relative index will change while interpreting instructions
 on the encoder stream.
 
 ~~~~~ drawing
-    +---+---------------+-----------+
-    | n |      ...      |   d + 1   |  Absolute Index
-    + - +---------------+ - - - - - +
-    | 0 |      ...      | n - d - 1 |  Relative Index
-    +---+---------------+-----------+
-      ^                       |
-      |                       V
-Insertion Point         Dropping Point
+      +---+---------------+-----------+
+      | n |      ...      |   d + 1   |  Absolute Index
+      + - +---------------+ - - - - - +
+      | 0 |      ...      | n - d - 1 |  Relative Index
+      +---+---------------+-----------+
+      ^                               |
+      |                               V
+Insertion Point                 Dropping Point
 
 n = count of entries inserted
 d = count of entries dropped
 ~~~~~
 {: title="Example Dynamic Table Indexing - Control Stream"}
 
 Because frames from request streams can be delivered out of order with
-instructions on the encoder stream, relative indices are relative to the Base
-Index at the beginning of the header block (see {{header-prefix}}). The Base
-Index is an absolute index. When interpreting the rest of the frame, the entry
-identified by Base Index has a relative index of zero.  The relative indices of
-entries do not change while interpreting headers on a request or push stream.
+instructions on the encoder stream, relative indices are relative to the Base at
+the beginning of the header block (see {{header-prefix}}). The Base is encoded
+as a value relative to the Insert Count, so it can be expressed as the number of
+entries inserted. The dynamic table entries up to the Base can be referenced.
+The most recently inserted entry within the limit set by the Base has a relative
+index of 0.
+
+Though new entries could be added while processing a header block, the relative
+indices of entries do not change.
```

Comment:
How are new entries added while processing a header block?  Do you mean on the encoder side?  Maybe encoding would be better than processing then?

Reply:
In theory, you could process instructions on the encoder stream in parallel with interpreting a header block.  However, more likely this refers to entries getting inserted during encoding, and this is the wrong terminology.

Reply:
I was referring to parallel processing.  My implementation allows for that possibility.